File: salon-frontend/src/app/core/services/staff-review.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { ApiResponse } from '../models/api-response.model';
import { StaffReviewModel, CreateStaffReviewRequest } from '../models/staff-review.model';

const API_URL = 'http://localhost:8080/api/v1';

@Injectable({ providedIn: 'root' })
export class StaffReviewService {
  private http = inject(HttpClient);

  createReview(request: CreateStaffReviewRequest): Observable<ApiResponse<StaffReviewModel>> {
    return this.http.post<ApiResponse<StaffReviewModel>>(`${API_URL}/staff-reviews`, request);
  }

  getReviewByAppointmentId(appointmentId: number): Observable<ApiResponse<StaffReviewModel>> {
    return this.http.get<ApiResponse<StaffReviewModel>>(`${API_URL}/staff-reviews/appointment/${appointmentId}`);
  }
}

File: salon-backend/src/main/java/com/salon/module/staff/service/StaffReviewService.java
package com.salon.module.staff.service;

import com.salon.module.staff.dto.StaffReviewRequest;
import com.salon.module.staff.dto.StaffReviewResponse;

public interface StaffReviewService {
    StaffReviewResponse createReview(Long userId, StaffReviewRequest request);
    StaffReviewResponse getReviewByAppointmentId(Long userId, Long appointmentId);
}
